tests/unit/client-eviction.tcl

```diff
@@ -0,0 +1,475 @@
+tags {"external:skip"} {
+
+# Get info about a redis client connection:
+# name - name of client we want to query
+# f - field name from "CLIENT LIST" we want to get
+proc client_field {name f} {
+    set clients [split [string trim [r client list]] "\r\n"]
+    set c [lsearch -inline $clients *name=$name*]
+    if {![regexp $f=(\[a-zA-Z0-9-\]+) $c - res]} {
+        throw no-client "no client named $name found with field $f"
+    }
+    return $res
+}
+
+# Sum a value across all redis client connections:
+# f - the field name from "CLIENT LIST" we want to sum
+proc clients_sum {f} {
+    set sum 0
+    set clients [split [string trim [r client list]] "\r\n"]
+    foreach c $clients {
+        if {![regexp $f=(\[a-zA-Z0-9-\]+) $c - res]} {
+            throw no-field "field $f not found in $c"
+        }
+        incr sum $res
+    }
+    return $sum
+}
+
+proc write_err_exception {e} {
+    return [regexp {(.*connection reset by peer.*|.*broken pipe.*)} $e]
```

Comment:
Another `error writing "sock11": protocol wrong type for socket` error caused the test to fail.

Reply:
Very weird. Looks like a MacOS kernel bug: http://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/
I'll adopt the regex accordingly.